### PR TITLE
#159082320 Add Search Functionality with custom filters

### PR DESCRIPTION
**Pull request by RutaleIvanPaul**
#### What does this PR do?
Adds functionality to enable users search for articles by using custom filters like title, author and tags.
#### Related Stories
#159082320

### CODE REVIEW

File: authors/apps/articles/tests/test_search_article.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    def test_search_by_tag_normal(self):
+            request = self.factory.get('api/articles/search?tag=test')
+            response = SearchArticlesAPIView.as_view()(request)
+            self.assertGreater(response.data["count"], 0)
+
+    def test_search_by_tag_non_existent(self):
+        request = self.factory.get('api/articles/search?tag=XYZ')
+        response = SearchArticlesAPIView.as_view()(request)
+        self.assertEqual(response.data["count"], 0)
```
Comment (pluwum) [inline authors/apps/articles/tests/test_search_article.py]: add a new line here @RutaleIvanPaul

File: authors/apps/articles/urls.py [hunk 2]
```diff
@@ -20,7 +19,5 @@
     path('articles/<slug>/like',LikeArticleAPIView.as_view()),
     path('articles/<slug>/dislike',DislikeArticleAPIView.as_view()),
     path('articles/<slug>/favorite',FavoriteArticleAPIView.as_view()),
-    path('my/articles/', ListAuthorArticlesAPIView.as_view()),
```
Comment (pluwum) [inline authors/apps/articles/urls.py]: Help me understand why these URLs are being removed.

**Comment by pluwum** [comment]
Rebase and resolve conflicts on this PR.

File: authors/apps/articles/urls.py (hunk 2 above)
Comment (RutaleIvanPaul) [reply authors/apps/articles/urls.py]: Those urls are no longer needed. The functionality they were providing is no longer necessary.

File: authors/apps/articles/tests/test_search_article.py (hunk 1 above)
Comment (RutaleIvanPaul) [reply authors/apps/articles/tests/test_search_article.py]: Alright

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -218,3 +222,4 @@ def put(self, request, *args, **kwargs):
             raise exceptions.APIException(e)
 
         return Response(data=self.serializer_class(updated_article).data, status=status.HTTP_201_CREATED)
+
```
Comment (pluwum) [inline authors/apps/articles/views.py:221]: Add a new line here

**Pull request merged**